torch/distributed/_tensor/sharding_prop.py

```diff
@@ -417,3 +436,42 @@ def _select_strategy(self, strategy: OpStrategy) -> PlacementStrategy:
 
         # for eager execution, we just select the one with the minimal redistribute cost
         return strategy.strategies[strategy_costs.index(min(strategy_costs))]
+
+    def _adjust_size_and_stride_args(
+        self, op_schema: OpSchema, spec: DTensorSpec, mesh: DeviceMesh
+    ) -> OpSchema:
+        expected_input_schema = list(op_schema.args_schema)
+        # hard code size_idx until we need to support different cases
+        size_idx = 1
+        size = expected_input_schema[size_idx]
+        assert isinstance(size, list)
+        size = list(size)
+        # adjust size to be the same as that of the _local_tensor
+        # of the DTensor input arg at index 0, which can only be inferred
+        for i, p in enumerate(spec.placements):
+            if p.is_shard():
+                assert isinstance(p, Shard)
+                size[p.dim] //= mesh.size(i)
+        expected_input_schema[size_idx] = torch.Size(size)
+
+        # adjust the stride arg for aten.new_empty_strided.default
+        if op_schema.op == aten.new_empty_strided.default:
+            # hard code stride_idx until we need to support different cases
+            stride_idx = 2
+            stride = expected_input_schema[stride_idx]
+            assert isinstance(stride, list)
+
+            stride_divisors = [1] * len(stride)
+            for i, mesh_idx in enumerate(spec.dim_map):
```

Comment:
can we make this a util function? I think this would be useful for some other features in the future, similar to this https://github.com/pytorch/pytorch/blob/main/torch/distributed/_tensor/_utils.py#L17

Also please not use `dim_map`, at least that's a field we would need to do some refactor